### PR TITLE
Add `utils_aws_eks_update_kubeconfig` datasource

### PR DESCRIPTION
**Pull request by aknysh**
## what
* Add `utils_aws_eks_update_kubeconfig` datasource

## why
* Download `kubeconfig` from EKS clusters and save it to a file using Terraform, then use the downloaded `kubeconfig` in the Terraform `helmfile` provider to provision helm charts with helmfiles

```
The data source can executes 'aws eks update-kubeconfig' commands in four different ways:

1. If all the required parameters (cluster name and AWS profile/role) are provided,
   then it executes the command without requiring the CLI config ('atmos.yaml') and component/stack/context.
   'atmos.yaml' is not required/needed in this case.

2. If 'component' and 'stack' are provided,
   then it executes the command using the atmos CLI config (see atmos.yaml) and the context by searching for the following settings:
     - 'components.helmfile.cluster_name_pattern' in 'atmos.yaml' CLI config (and calculates the '--name' parameter using the pattern)
     - 'components.helmfile.helm_aws_profile_pattern' in 'atmos.yaml' CLI config (and calculates the '--profile' parameter using the pattern)
     - 'components.helmfile.kubeconfig_path' in 'atmos.yaml' CLI config
     - the variables for the component in the provided stack
     - 'region' from the variables for the component in the stack

3. If the context ('tenant', 'environment', 'stage') and 'component' are provided,
   then it builds the stack name by using the 'stacks.name_pattern' CLI config from 'atmos.yaml', then performs the same steps as example #2.

4. Combination of the above. Provide a component and a stack (or context), and override other parameters (e.g. 'kubeconfig', 'region').

If 'kubeconfig' (the filename to write the kubeconfig to) is not provided, then it's calculated by joining
the base path from 'components.helmfile.kubeconfig_path' CLI config from 'atmos.yaml' and the stack name.

Supported inputs of the 'utils_aws_eks_update_kubeconfig' data source:
  - component
  - stack
  - tenant
  - environment
  - stage
  - cluster_name
  - kubeconfig
  - profile
  - role_arn
  - alias
  - region
```
## references
* https://github.com/cloudposse/atmos/pull/136
* https://docs.aws.amazon.com/cli/latest/reference/eks/update-kubeconfig.html

### CODE REVIEW

File: examples/complete/components/terraform/test/test-component/tenant1-ue2-dev-test-test-component-override-3.terraform.tfvars.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (nitrocode) [inline examples/complete/components/terraform/test/test-component/tenant1-ue2-dev-test-test-component-override-3.terraform.tfvars.json:1]: Should this be committed ?

**Review by nitrocode — APPROVED** [review]

File: examples/complete/components/terraform/test/test-component/tenant1-ue2-dev-test-test-component-override-3.terraform.tfvars.json (hunk 1 above)
Comment (aknysh) [reply examples/complete/components/terraform/test/test-component/tenant1-ue2-dev-test-test-component-override-3.terraform.tfvars.json]: thanks, deleted

Comment (aknysh) [reply examples/complete/components/terraform/test/test-component/tenant1-ue2-dev-test-test-component-override-3.terraform.tfvars.json]: @nitrocode please re-review

**Comment by nitrocode** [comment]
Can we add `*.terraform.tfvars.json` and `*.helmfile.vars.yaml` to the `.gitignore` as well ?

**Comment by aknysh** [comment]
@nitrocode updated gitignore and dockerignore

**Review by nitrocode — APPROVED** [review]

**Pull request merged**